src/js/player.js

```diff
@@ -2777,18 +2782,17 @@ class Player extends Component {
    * @listens keydown
    */
   handleKeyPress(event) {
+    if (!this.options_.userActions ||
+        this.options_.userActions.hotkeys === false) {
+      this.handleHotkeys(event);
+    } else if (typeof this.options_.userActions.hotkeys === 'function') {
 
-    if (this.options_.userActions && this.options_.userActions.hotkeys && (this.options_.userActions.hotkeys !== false)) {
-
-      if (typeof this.options_.userActions.hotkeys === 'function') {
+      this.options_.userActions.hotkeys.call(this, event);
 
-        this.options_.userActions.hotkeys.call(this, event);
-
-      } else {
+    } else {
 
-        this.handleHotkeys(event);
+      this.handleHotkeys(event);
```

Comment:
I don't follow why `this.handleHotkeys(event)` is called in this `else` clause; I don't see a flow through here which *doesn't* either call `handleHotkeys(event)` or `this.options_.userActions.hotkeys.call(this, event)`.

Reply:
might be easier in split view, but basically, if `userActions` isn't set or if `hotkeys` is false, call `handleHotkeys(events)` which has defaults for hotkeys. Otherwise, if hotkeys is a function, call it or call handleHotkeys.
Previously, if userActions wasn't set at all, handleHotkeys was never called.

Reply:
Yeah, "if userActions wasn't set at all, handleHotkeys was never called." - that was intentional. The default behavior for older setups was that this new hotkey handling isn't called, so that it doesn't conflict with the hotkeys plugin. Otherwise, if `userActions` *is* set (because some other option within `userActions` is being set, for some other feature), then setting `hotkeys` to false disabled this new hotkeys handling as well. If `hotkeys` is not set to false, then this new hotkey handling is used, unless `hotkeys` is set to a function, in  which case call that function *instead* of the video.js `handleHotkeys` method.

Reply:
My thought was to have it off by default for now, so we don't need to explicitly check for the hotkeys plugin, which seems like a hack. Eventually we should consider integrating more/all of the features of the hotkeys plugin and turning on something by default, but we don't need to rush to that - it's not an a11y requirement to have hotkeys, it's just a usability benefit. My aim with this change was to implement the hotkeys in a way that doesn't *break* accessibility.

Reply:
https://github.com/videojs/video.js/issues/5765